File: fizzbuzz.py

cannot_run = True if __name__ != '__main__' else False



def calculate(number, condition_number, result_word, current_output):

    # <Guard Clauses>
    problem = type(number) != int
    problem = type(condition_number) != int
    problem = type(result_word) != str
    problem = type(current_output) != str
    problem = len(result_word.split()) != 1
    if problem: raise SyntaxError('Invalid Arguments for calculate')
    # </Guard Clauses>

    # <Capitalisation Chech>
    if current_output == '': word = result_word[0].upper() + result_word[1:].lower()
    else: word = result_word.lower()
    # </Capitalisation Check>

    # <Condition Check>
    if number % condition_number == 0: return word
    else: return ''
    # </Condition Check>



def main():
    
    if cannot_run: raise RuntimeError('Main Called from non Entry Point')

    ITERATIONS = 100

    for number in range(1, ITERATIONS + 1):
        
        output = ''
        output += calculate(number, 3, 'fizz', output)
        output += calculate(number, 5, 'buzz', output)
        if output == '': output = number
        
        print(output)

main()

File: fizzbuzz.c
#include <stdio.h>
#include <ctype.h>
#include <string.h>

void strclear(char string[]) { strcpy(string, ""); }

void strcap(char string[]) {

	char output_string[500];
	for ( int i = 0; i <= strlen(string); i++ ) {
		
		if (i == 0) {output_string[i] = toupper(string[i]);}
		else {output_string[i] = tolower(string[i]);}
	
	}

	strcpy(string, output_string);

}

void calculate(int number, int condition_number, char result_word[], char output[]) {
	
	int is_first = 0;

	if ( strlen(output) == 0 ) { is_first = 1; }
	if ( number % condition_number == 0 ) { strcat(output, result_word); }
	if ( is_first == 1 ) { strcap(output); }

}

void main() {

	char output[500] = "";
	int ITERATIONS = 100;

	for ( int number = 1; number <= ITERATIONS; number++ ) {
	
		strclear(output);
		calculate(number, 3, "fizz", output);
		calculate(number, 5, "buzz", output);
		
		if ( strlen(output) == 0 ) { printf("%d\n", number); }
		else { printf("%s\n", output); }

	}

}
